### PR TITLE
Add `device` keyword to creation functions

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

This PR adds `device` keyword to creation functions to conform with Array API (Only accepts `"cpu"` now).

### CODE REVIEW

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Thanks, LGTM besides the one comment.

File: src/finch/tensor.py [hunk 1]
```diff
@@ -683,34 +686,53 @@ def full_like(
     *,
     dtype: DType | None = None,
     format: str = "coo",
+    device: Literal["cpu"] | None = None,
```
Comment (hameerabbasi) [inline src/finch/tensor.py]:
```suggestion
    device: Device = None,
```

And let's define `Device` in `finch.typing`?

Comment (mtsokol) [reply src/finch/tensor.py]: Done!

**Review by hameerabbasi — APPROVED** [review]

**Comment by willow-ahrens** [comment]
Finch does have a notion of device, though it's not quite ready for production. You can view it in https://github.com/willow-ahrens/Finch.jl/blob/main/src/architecture.jl

**Comment by willow-ahrens** [comment]
Maybe once that's up and running we can port it to python somehow, but I think we'll need to discuss some more.

**Pull request merged**